ip/ip.go

```diff
@@ -15,6 +16,27 @@ type addr struct {
 }
 
 func GetIpAddress(vmName string) (string, error) {
+
+	health, err := hypervisor.HealthCheck(vmName, "status")
+	if err != nil {
+		return "", err
+	}
+
+	// Wait for health status to be "Up" for up to 5 minutes
+	timeout := time.After(5 * time.Minute)
```

Comment:
I like this piece of code, thanks for it.

but as I mentioned, I'd prefer this code to encapsulated and called as part of `GetVMProperty`

if you think about it, not "status" property nor the "Up" response string is global across hypervisors.